src/m365/aad/commands/group/group-remove.spec.ts

```diff
@@ -0,0 +1,158 @@
+    });
+    promptOptions = undefined;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.delete,
+      aadGroup.getGroupIdByDisplayName,
+      Cli.prompt
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.service.connected = false;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.GROUP_REMOVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('removes the specified group by id without prompting for confirmation', async () => {
+    const deleteRequestStub = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { verbose: true, id: groupId, force: true } });
+    assert(deleteRequestStub.called);
+  });
+
+  it('removes the specified group by displayName while prompting for confirmation', async () => {
+    sinon.stub(aadGroup, 'getGroupIdByDisplayName').resolves(groupId);
+
+    const deleteRequestStub = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').resolves({ continue: true });
+
+    await command.action(logger, { options: { verbose: true, displayName: displayName } });
+    assert(deleteRequestStub.called);
+  });
+
+  it('throws an error when group by id cannot be found', async () => {
+    const error = {
+      error: {
+        code: 'Request_ResourceNotFound',
+        message: `Resource '${groupId}' does not exist or one of its queried reference-property objects are not present.`,
+        innerError: {
+          date: '2023-08-30T14:32:41',
+          'request-id': 'b7dee9ee-d85b-4e7a-8686-74852cbfd85b',
+          'client-request-id': 'b7dee9ee-d85b-4e7a-8686-74852cbfd85b'
+        }
+      }
+    };
+    sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/groups/${groupId}`) {
+        throw error;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, { options: { verbose: true, id: groupId, force: true } }),
+      new CommandError(error.error.message));
+  });
+
+  it('prompts before removing the specified group setting when confirm option not passed', async () => {
+    await command.action(logger, { options: { id: groupId } });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the group setting when prompt not confirmed', async () => {
+    const deleteSpy = sinon.spy(request, 'delete');
```

Comment:
Let's convert this to a stub that always resolves.
If we ever break this test and we perform a DELETE request while we cancel the prompt, we will get a timeout because the delete request is not stubbed, making it harder to understand why the test failed.
If we just stub the request, instead of a timeout, we will receive a validation error in the assert function which will be much clearer.